protocol/meson.build

```diff
@@ -1,103 +1,147 @@
-wl_protocol_dir = wayland_protos.get_pkgconfig_variable('pkgdatadir')
+wl_protocol_dir = wayland_protos.get_variable(pkgconfig: 'pkgdatadir')
 
 wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)
 if wayland_scanner_dep.found()
 	wayland_scanner = find_program(
-		wayland_scanner_dep.get_pkgconfig_variable('wayland_scanner'),
+		wayland_scanner_dep.get_variable(pkgconfig: 'wayland_scanner'),
 		native: true,
 	)
 else
 	wayland_scanner = find_program('wayland-scanner', native: true)
 endif
 
-protocols = [
-	[wl_protocol_dir, 'stable/presentation-time/presentation-time.xml'],
-	[wl_protocol_dir, 'stable/xdg-shell/xdg-shell.xml'],
-	[wl_protocol_dir, 'unstable/fullscreen-shell/fullscreen-shell-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/idle-inhibit/idle-inhibit-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/pointer-constraints/pointer-constraints-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/pointer-gestures/pointer-gestures-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/primary-selection/primary-selection-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/relative-pointer/relative-pointer-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/tablet/tablet-unstable-v2.xml'],
-	[wl_protocol_dir, 'unstable/text-input/text-input-unstable-v3.xml'],
-	[wl_protocol_dir, 'unstable/xdg-decoration/xdg-decoration-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/xdg-output/xdg-output-unstable-v1.xml'],
-	[wl_protocol_dir, 'unstable/xdg-shell/xdg-shell-unstable-v6.xml'],
-	'gtk-primary-selection.xml',
-	'idle.xml',
-	'input-method-unstable-v2.xml',
-	'server-decoration.xml',
-	'virtual-keyboard-unstable-v1.xml',
-	'wlr-data-control-unstable-v1.xml',
-	'wlr-export-dmabuf-unstable-v1.xml',
-	'wlr-foreign-toplevel-management-unstable-v1.xml',
-	'wlr-gamma-control-unstable-v1.xml',
-	'wlr-input-inhibitor-unstable-v1.xml',
-	'wlr-layer-shell-unstable-v1.xml',
-	'wlr-output-management-unstable-v1.xml',
-	'wlr-screencopy-unstable-v1.xml',
-]
+protocols = {
+	# Stable upstream protocols
+	'xdg-shell': {
+		'path': wl_protocol_dir / 'stable/xdg-shell/xdg-shell.xml',
+		'client': true,
```

Comment:
I wonder if it's worth it to worry about generating client code unnecessarily. As I understand it, the target will only get built if something else depends on it (even if it would get built unconditionally it wouldn't be a big deal).

We could remove the `client` field.

Reply:
Done. I just used build_by_default: false

Reply:
I guess it's already the default since we require Meson >= 0.51. It doesn't hurt to explicitly set it though.

> the default value is false (changed in 0.50)